docs/src/modules/components/MarkdownDocs.js

```diff
@@ -89,10 +90,25 @@ const styles = (theme) => ({
 });
 
 function MarkdownDocs(props) {
-  const { classes, disableAd = false, disableToc = false, demos, docs, requireDemo } = props;
+  const {
+    classes,
+    disableAd = false,
+    disableToc = false,
+    demos: demosProp,
+    docs: docsProp,
+    requireDemo,
+    pageFilename,
+    requireRaw,
+  } = props;
 
-  const t = useSelector((state) => state.options.t);
+  let demos = demosProp;
+  let docs = docsProp;
+
+  if (process.env.NODE_ENV !== 'production' && pageFilename && requireRaw) {
```

Comment:
This should not be in. Not having hot-reload for getInitialProps is a nextjs issue not Material-UI issue. This is creates fudamentally different dev and prod environments for little gain. 

Live markdown is available directly in vscode and I'd be surprised if  other IDEs don't have an extension for markdown as well.

This sets a dangerous precedent for monkey-patching hot-releoading for every getInitialProps, getStatisProps, getStaticPaths etc.

Eventually this won't work anyway once we move from getInitialProps to getStaticProps which might result in nonde-only code.

Reply:
Ok, let's do it incrementally. I'm removing the change. Let's see how the DX is in the coming days. I guess it's a matter of habit. I always end-up testing how it looks like for the users. Is the text too long, is the layout correct, etc.

Reply:
@timneutkens Has the Next.js team considered to support live reload on `getInitialProps`? Related to https://twitter.com/timneutkens/status/1251942933953228802.

Reply:
> @timneutkens Has the Next.js team considered to support live reload on `getInitialProps`/`getStatisProps`? Related to [twitter.com/timneutkens/status/1251942933953228802](https://twitter.com/timneutkens/status/1251942933953228802).

Yes! Won't be part of the fast refresh work but definitely something we want to tackle.